src/server/server.js

```diff
@@ -197,6 +197,11 @@ Server.prototype._onSentryDown = function (sentryId) {
   var started = Date.now()
 
   nonblocking(presences).forEach(function (presence) {
+    // check if presence was destroyed
+    if (!presence || !presence.manager || !presence.manager.store) {
```

Comment:
Ok, this makes sense, thank you for the thorough explanation. The piece I was missing is that it's okay for us not to process sentry down messages for presences with no remaining subscribers, because the only potential audience for the synthetic offline messages is those same subscribers. Filtering out those presence resources here allows us to short circuit processing those Presences before trying to `broadcast` to an audience of 0 subscribers later on.

I've updated the PR to add a `Resource#destroyed` flag and check for this in the sentry down handler.